### PR TITLE
Flat pages detail

### PR DESCRIPTION
**Pull request by zatan**
https://favro.com/card/b03d6d6d9b11b61167ac4246/Dev-2782

### CODE REVIEW

File: apps/pages/urls.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from django.conf.urls import url
```
Comment (tomkins) [inline apps/pages/urls.py]: Don't think you need this?

File: commonslibrary/urls.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'^accounts/', include('accounts.urls')),
+    url(r'pages/', include('pages.urls')),
```
Comment (tomkins) [inline commonslibrary/urls.py]: Don't think you need this either?

File: apps/pages/models.py [hunk 3]
```diff
@@ -48,3 +49,6 @@ class Meta:
 
     def __str__(self):
         return "{url} -- {title}".format(url=self.url, title=self.title)
+
+    def get_absolute_url(self):
+        return reverse('pages:page-detail', kwargs={'url': self.url.lstrip('/')})
```
Comment (tomkins) [inline apps/pages/models.py]: Think this could be `self.url`?

File: apps/pages/urls.py (hunk 1 above)
Comment (zatan) [reply apps/pages/urls.py]: Why don't I need it?

File: commonslibrary/urls.py (hunk 2 above)
Comment (zatan) [reply commonslibrary/urls.py]: Why? I follow the way django.contrib.flatpages.urls does.

File: apps/pages/models.py (hunk 3 above)
Comment (zatan) [reply apps/pages/models.py]: No, as it adds two slashes

**Comment by zatan** [comment]
It's handled by 404 and I wanted to keep urls.py and the middleware as I follow the way django works with the flatpages am I missing something here? `/pages/` `/url`  I want both these cases for the urls.

**Comment by tomkins** [comment]
The instructions for flatpages: https://docs.djangoproject.com/en/1.11/ref/contrib/flatpages/#installation

Suggests either editing the URLconf, or adding the middleware - not both.

**Comment by zatan** [comment]
Updated

**Comment by zatan** [comment]
Been merged.

**Pull request closed without merging**